lib/ui/window.dart

```diff
@@ -150,7 +150,37 @@ class Locale {
   /// Registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry).
   /// Typically this means the primary language subtag should be lowercase and
   /// the region subtag should be uppercase.
-  const Locale(this._languageCode, [ this._countryCode ]) : assert(_languageCode != null);
+  const Locale(
+    this._languageCode, [
+    this._countryCode,
+  ]) : assert(_languageCode != null),
+       this.scriptCode = null,
+       this._variants = null;
+
+  /// Creates a new Locale object.
+  ///
+  /// The keyword arguments specify the subtags of the Locale. The subtag values
+  /// should match the values listed as valid in Unicode CLDR supplemental data:
+  /// [language](http://unicode.org/cldr/latest/common/validity/language.xml),
+  /// [script](http://unicode.org/cldr/latest/common/validity/script.xml),
+  /// [region](http://unicode.org/cldr/latest/common/validity/region.xml) and
+  /// [variant](http://unicode.org/cldr/latest/common/validity/variant.xml).
+  /// There may be multiple variants as long as they appear in sorted order and
+  /// are joined by hyphens.
+  const Locale.fromComponents({
+    String language,
+    String script,
+    String region,
+    String variants,
+  }) : assert(language == null || (language.length >= 2 && language.length <= 8
+                                   && language.length != 4)),
+       assert(script == null || script.length == 4),
+       assert(region == null || (region.length >= 2 && region.length <= 3)),
+       assert(variants == null || variants.length >= 4),
+       _languageCode = language ?? 'und',
```

Comment:
nit: this should just be the default value of the argument (that makes the code more self-documenting)

Reply:
Thanks!